api/src/integrationTest/kotlin/com/cosmotech/api/home/workspace/WorkspaceControllerTests.kt

```diff
@@ -658,4 +659,40 @@ class WorkspaceControllerTests : ControllerTestBase() {
             document(
                 "organizations/{organization_id}/workspaces/{workspace_id}/files/download/GET"))
   }
+
+  @Test
+  @WithMockOauth2User
+  fun `download workspace file with wrong file name`() {
+
+    val workspaceId =
+        createWorkspaceAndReturnId(
+            mvc, organizationId, constructWorkspaceCreateRequest(solutionId = solutionId))
+
+    val fileName = "test.txt"
+    val fileToUpload =
+        this::class.java.getResourceAsStream("/workspace/$fileName")
+            ?: throw IllegalStateException(
+                "$fileName file used for organizations/{organization_id}/workspaces/{workspace_id}/files/POST endpoint documentation cannot be null")
+
+    val mockFile =
+        MockMultipartFile(
+            "file", fileName, MediaType.TEXT_PLAIN_VALUE, IOUtils.toByteArray(fileToUpload))
+
+    val destination = "path/to/a/directory/"
+    mvc.perform(
+        multipart("/organizations/$organizationId/workspaces/$workspaceId/files")
+            .file(mockFile)
+            .param("overwrite", "true")
+            .param("destination", destination)
+            .accept(MediaType.APPLICATION_JSON)
+            .with(csrf()))
+
+    mvc.perform(
+            get("/organizations/$organizationId/workspaces/$workspaceId/files/download")
+                .param("file_name", "Wrong file name")
+                .accept(MediaType.APPLICATION_OCTET_STREAM))
+        .andExpect(status().is4xxClientError)
+        .andExpect(jsonPath("$.detail").value("The specified key does not exist."))
```

Comment:
The user probably has no idea what a "key" is. User facing errors should refer to user facing concepts, here "file name". But that probably means catching and handling the error differently, maybe this is not easy. And we probably have similar pattern elsewhere anyway...